src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -269,14 +269,21 @@ ray::Status OwnershipBasedObjectDirectory::SubscribeObjectLocations(
     auto msg_published_callback = [this, object_id](const rpc::PubMessage &pub_message) {
       RAY_CHECK(pub_message.has_worker_object_locations_message());
       const auto &location_info = pub_message.worker_object_locations_message();
-      ObjectLocationSubscriptionCallback(location_info, object_id,
-                                         /*location_lookup_failed*/ false);
+      ObjectLocationSubscriptionCallback(
+          location_info, object_id,
+          /*location_lookup_failed*/ !location_info.ref_removed());
+      if (location_info.ref_removed()) {
+        mark_as_failed_(object_id, rpc::ErrorType::OBJECT_RELEASED);
```

Comment:
I guess this is probably not necessary? This APIs are used to pull new objects, and I feel like this error should be already propagated by the higher level layer (in the future resolution)? I also have seen this this happens A LOT, and kind of worried that it will create so many synchronous calls to plasma store.

Reply:
Did you happen to find when this is needed?

Reply:
I think we do want to throw the error because it can be a real reference counting issue. But I agree we shouldn't log WARNINGs.